asv.conf.json

```diff
@@ -4,20 +4,19 @@
     "project_url": "https://qiskit.org",
     "repo": ".",
     "install_command": [
-        "in-dir={env_dir} python -mpip install {wheel_file}[all] python-constraint qiskit-experiments==0.3.0"
+        "in-dir={env_dir} python -m pip install {wheel_file}[all] python-constraint qiskit-experiments==0.3.0"
```

Comment:
Do you think it makes sense to have a matrix here for the experiments version and python-constraint: https://asv.readthedocs.io/en/stable/asv.conf.json.html#matrix? This pattern was used in the original asv benchmarks that I added back in 2019 but back then asv didn't have support for reasoning about third-part dependencies explicitly like this, so having a pinned install in the install command was the only pattern to enable this.

Reply:
I didn't know about that, but it looks good.  It gets potentially hairy with `qiskit-experiments`, because that has a transitive dependency on Terra, and matrix requirements apparently get installed before the main project.  The current form forces version resolution to happen all at once, which is a bit more stable, I suppose.